lib/models/workdir-cache.js

```diff
@@ -56,6 +56,10 @@ export default class WorkdirCache {
       let currentDir = initialDir;
 
       const check = () => {
+        if (!isValidWorkdir(currentDir)) {
+          return walk();
+        }
```

Comment:
:+1: I like that we're skipping `~` and `/` when we're traversing directories. Why even run stat if we're just going to throw it out.